### ISSUE TITLE
[HOLD 53493] [$250] iOS - Attachment - No sliding in animation when opening attachment

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#53362]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.69-1
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Y
**Email or phone of affected tester (no customers):** applausetester+zm63@applause.expensifail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Launch ND or hybrid app.
2. Go to DM.
3. Tap on the chat header.
4. Tap on the user avatar.
5. Note that there is no sliding in animation when opening avatar.
6. Submit an expense with a receipt.
7. Send an image to the chat.
8. Open the receipt from the expense.
9. Open the image attachment.

## Expected Result:
There will be sliding in animation when opening attachment (production behavior).

## Actual Result:
There is no sliding in animation when opening attachment. The app blinks slightly when the attachment is opened.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/01669cc4-aeab-42b4-bbcb-b9835be86f19
  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021863486574634374199</li>
        <li>Upwork Job ID: 1863486574634374199</li>
        <li>Last Price Increase: 2024-12-02</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Reviewer | 105182252</li>
</ul></li>
    </ul>
</details>

**Comment by mountiny** [comment]
Thai must be coming from the native stack change 

@chrispader @hannojg @kirillzyusko for eyes

I dont think this is a blocker as users can open and view the attachment fine, teated on ios too

**Comment by Christinadobrzyn** [comment]
Catching up here - I think we're waiting on proposals, is that right @ikevin127?

**Comment by ikevin127** [comment]
Yes, unless somebody from our expert agencies wants to handle this - we're open for proposals based on the context mentioned in https://github.com/Expensify/App/issues/53362#issuecomment-2510775039.

**Comment by mountiny** [comment]
Margelo will handle this one

**Comment by hannojg** [comment]
This was scheduled to @chrispader who already started looking into it, can you pls comment @chrispader. Thanks!

**Comment by chrispader** [comment]
Yes, i'm looking into it right now

**Comment by chrispader** [comment]
I've created a PR with a fix for this specific issue here: https://github.com/Expensify/App/pull/53475

This fix is a hacky workaround tbh, since we're basically just delaying the navigation by 200ms.

The problem we see in this issue is the same as previous problems we had with RN modals/`react-native-modal`s in the NativeStack PR. When modals are mounted/opened on a new screen - essentially at the same the time the screen mounts - the slide in animation will not be visible. Same thing for unmounting a modal. If the modal is hidden/unmounted at the same time as we navigate back from a screen, the animation cannot be seen.

This should ultimately by fixed with a more holistic solution, where we replace all legacy modals with modal (screens) from `@react-navigation`, though i think this should be handled in a separate PR.

For example in this issue, the problem is with the `AttachmentModal` component used in `ProfileAvatar`. It is used in many places. Same thing goes for other screens, that are essentially just modals. This won't work with NativeStack.

**Comment by Christinadobrzyn** [comment]
Thanks @chrispader - I added you to this GH. @mountiny and @ikevin127 does this solution sound good? https://github.com/Expensify/App/issues/53362#issuecomment-2514954175

**Comment by Christinadobrzyn** [comment]
Monitoring PR https://github.com/Expensify/App/pull/53475

**Comment by ikevin127** [comment]
PR was reverted as it had 2 regressions, we're probably looking at a re-work. No compensation required since the PR was reverted.

**Comment by ikevin127** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by Christinadobrzyn** [comment]
Looks like we're ready for payment but I don't see anything about a regression test. 

@ikevin127 do we need a regression test for this? 

Contributor+: @ikevin127  owed $250 via Upwork (https://www.upwork.com/nx/wm/offer/105182252)

**Comment by ikevin127** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by ikevin127** [comment]
> PR was reverted as it had 2 regressions, we're probably looking at a re-work. No compensation required since the PR was reverted.

@Christinadobrzyn No payment required here as the PR was reverted. Not sure where we are with the issue, maybe @chrispader has more context on what happened after the revert regarding the issue.

**Comment by Christinadobrzyn** [comment]
Ah thank you @ikevin127 - really appreciate the clarity. @chrispader can you let us know your thoughts on next steps?

**Comment by mountiny** [comment]
I think this still has to be handled, gonna make it weekly over the holidays

**Comment by chrispader** [comment]
This issue would get fixed by https://github.com/Expensify/App/issues/53493, so i'm going to further push forward the modal migration solution

**Comment by Christinadobrzyn** [comment]
Thanks for the update @chrispader! We'll add a HOLD to this and monitor - https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
hold for https://github.com/Expensify/App/issues/53493 - moving this to weekly.

**Comment by Christinadobrzyn** [comment]
hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Hold for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Hold for https://github.com/Expensify/App/issues/53493

Moving this to monthly because I'll be ooo next week. When I'm back on April 7th, I'll move it to weekly.

**Comment by ikevin127** [comment]
♻ Still on HOLD for https://github.com/Expensify/App/issues/53493.

**Comment by ikevin127** [comment]
♻ Still on HOLD for https://github.com/Expensify/App/issues/53493.

**Comment by Christinadobrzyn** [comment]
Still on HOLD for https://github.com/Expensify/App/issues/53493.

**Comment by Christinadobrzyn** [comment]
Still on HOLD for https://github.com/Expensify/App/issues/53493.

**Comment by Christinadobrzyn** [comment]
Still on HOLD for https://github.com/Expensify/App/issues/53493.

**Comment by Christinadobrzyn** [comment]
Still on HOLD for https://github.com/Expensify/App/issues/53493 - should we move this to monthly?

**Comment by Christinadobrzyn** [comment]
Still on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Still on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
Still on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493

**Comment by Christinadobrzyn** [comment]
on HOLD for https://github.com/Expensify/App/issues/53493 this is still on hold right @ikevin127 @chrispader?

**Comment by mountiny** [comment]
@chrispader I think this might be off hold, is that right?

**Comment by chrispader** [comment]
@Christinadobrzyn yes indeed, this issue has been fixed by my previous PRs. Here's a screen recording of the current behavior on `main`

https://github.com/user-attachments/assets/d847ec50-e96e-455a-8cb5-29329d5c7a31

So this can be closed! 🙌🏼
